exchangelib/folders.py

```diff
@@ -117,6 +117,12 @@ def __init__(self, id, changekey):
         self.id = id
         self.changekey = changekey
 
+    @property
+    def item_id(self):
```

Comment:
I would prefer not to add this. It quickly leads to code like` item_id.item_id` which is confusing. Also, it adds possibly costly memory to an object that we may create millions of (I haven't tested this, though).

AFAICS you are only using this in tests to pass `ids` on to services. Policy until now is to let services support both `tuple`, `Item` and `ItemId` objects as input transparently where possible. I'll leave a comment in the test code.